test/extended/operators/check_crd_status.go

```diff
@@ -0,0 +1,127 @@
+package operators
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	"k8s.io/kube-openapi/pkg/util/sets"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apiextensionsclientset "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// checkCRDs returns a list of names of CRDs that don't have a "status" in the CRD schema
+// and also a subresource.status defined.
+// For now, it ignores the ones that are currently failing.
+func checkCRDs(crdItemList []apiextensionsv1.CustomResourceDefinition) []string {
+
+	// These CRDs, at the time this test was written, do not have a "status" in the CRD schema
+	// and subresource.status.
+	// These can be skipped for now but we don't want the number to increase.
+	// These CRDs should be tidied up over time.
+	//
+	exceptionsList := sets.NewString(
+		"builds.config.openshift.io",
+		"clusternetworks.network.openshift.io",
+		"consoleclidownloads.console.openshift.io",
+		"consoleexternalloglinks.console.openshift.io",
+		"consolelinks.console.openshift.io",
+		"consolenotifications.console.openshift.io",
+		"consoleplugins.console.openshift.io",
+		"consolequickstarts.console.openshift.io",
+		"consoleyamlsamples.console.openshift.io",
+		"egressnetworkpolicies.network.openshift.io",
+		"hostsubnets.network.openshift.io",
+		"imagecontentpolicies.config.openshift.io",
+		"imagecontentsourcepolicies.operator.openshift.io",
+		"machineconfigs.machineconfiguration.openshift.io",
+		"netnamespaces.network.openshift.io",
+		"networks.config.openshift.io",
+		"networks.operator.openshift.io",
+		"operatorpkis.network.operator.openshift.io",
+		"profiles.tuned.openshift.io",
+		"rangeallocations.security.internal.openshift.io",
```

Comment:
does not have a status field.

Something isn't right if you have to make exceptions for cases without status in the schema.

Reply:
Agreed.  But I thought the original intent of this jira/PR was to ensure that it doesn't get worse -- has that changed?